scraper.rb

```diff
@@ -18,52 +21,23 @@ def date_from(text)
   Date.parse(text).to_s rescue ''
 end
 
+def scrape(h)
+  url, klass = h.to_a.first
+  klass.new(response: Scraped::Request.new(url: url).response)
+end
+
 def scrape_list(url)
-  noko = noko_for(url)
-  noko.css('.fusion-portfolio-post').each do |p|
-    party_id, party = p.css('.fusion-portfolio-content h4').text.tidy.split(':', 2)
-    data = {
-      id:       p.attr('class')[/post-(\d+)/, 1],
-      name:     p.css('h2').text.tidy,
-      party_id: party_id,
-      party:    party,
-      email:    p.css('a[href*="@"]/@href').text.sub('mailto:', ''),
-      term:     8,
-      image:    p.css('.fusion-image-wrapper img/@src').text,
-      source:   p.css('h2 a/@href').text,
-    }
-    data.merge! scrape_person(data[:source])
-    ScraperWiki.save_sqlite(%i(id term), data)
+  members_page = scrape url => MembersPage
+  members_page.member_rows.each do |member|
+    ScraperWiki.save_sqlite(%i(id term), member.to_h
+                                               .merge(term: 8)
+                                               .merge((scrape member.source => MemberPage).to_h))
```

Comment:
The argument to `save_sqlite` is a bit too complex here. Building it up into a variable first is both cleaner, and also lets us print it out or examine it in `pry`, when debugging.